backend/packages/Upgrade/src/api/services/CacheService.ts

```diff
@@ -17,25 +17,32 @@ export class CacheService {
   }
 
   public setCache<T>(id: string, value: T): Promise<T> {
-    return this.memoryCache.set(id, value);
+    return this.memoryCache ? this.memoryCache.set(id, value) : Promise.resolve(null);
   }
 
   public getCache<T>(id: string): Promise<T> {
-    return this.memoryCache.get(id);
+    return this.memoryCache ? this.memoryCache.get(id) : Promise.resolve(null);
   }
 
-  public delCache<T>(id: string): Promise<T> {
-    return this.memoryCache.del(id);
+  public delCache(id: string): Promise<void> {
+    return this.memoryCache ? this.memoryCache.del(id) : Promise.resolve();
   }
 
-  public resetCache(): void {
-    this.memoryCache.reset();
+  public async resetPrefixCache(prefix: string): Promise<void> {
+    const keys = this.memoryCache ? await this.memoryCache.store.keys() : [];
+    const filteredKeys = keys.filter((str) => str.startsWith(prefix));
+    this.memoryCache.store.del(filteredKeys);
```

Comment:
I need a list of situations to test to make sure cache gets invalidated as expected for both valid experiments and segments caching.

- On experiment creation
- On experiment delete
- On experiment modify

- On segment creation
- On segment delete
- On segment modify

Are these the situations to check? Any others that are handled differently but would still need cache cleared?

Reply:
@danoswaltCL Earlier only segments were getting cached and resetting the cache would have not caused problems. But now we have two different types of cache so they need to be invalidated in a proper way. We should add a test case for cache invalidation. I suggest a separate PR for that.

Reply:
@danoswaltCL the above list looks good
CRUD for experiments (State change for experiment)
CRUD for segments

Reply:
@danoswaltCL In this portion of the code, I was supposed to clear the cached data. However, during testing, I mistakenly left in `console.log` statements instead of the deletion code. This commit rectifies this oversight. Yes, the commit message may not be entirely clear, rest assured that the code has been refined and is ready to go.